File: 프로그래머스/level2/42839.py
def check_decimal(num):
  for i in range(2, int(num**(1/2)) + 1):
    if num % i == 0:
      return False
  return True

def comb(d, arr, tmp, used, result):
  if d >= 1 and int(tmp) not in [0, 1]:
    result.add(int(tmp))
  if d == len(arr):
    return result
  for i in range(len(arr)):
    if used & (1 << i): continue
    result = comb(d + 1, arr, tmp + arr[i], used | (1 << i), result)
  return result

def solution(numbers):
  result = 0
  num_set = comb(0, numbers, '', 0, set())
  for n in num_set:
    result += check_decimal(n)
  return result

print(solution('17'))

File: 프로그래머스/level2/42839.js
const checkDecimal = (num) => {
  const sqrtNum = parseInt(Math.sqrt(num))
  for (let i = 2; i <= sqrtNum; i++) {
    if (num % i === 0) return false
  }
  return true
}

const makeNumber = (depth, iter, tmpNum, used, result) => {
  const num = parseInt(tmpNum) || 0
  if (depth >= 1 && num >= 2 && !result.includes(num)) {
    result.push(num)
  }
  if (depth === iter.length) {
    return result
  }
  for (let i = 0; i < iter.length; i++) {
    if (used & (1 << i)) {
      continue
    }
    result = makeNumber(depth + 1, iter, tmpNum + iter[i], used | (1 << i), result)
  }
  return result
}

const solution = (numbers) => (
  makeNumber(0, numbers, '', 0, [])
    .reduce((acc, cur) => acc + checkDecimal(cur), 0)
)

console.log(solution('17'))
console.log(solution('011'))

